### PR TITLE
[REF] web: convert views>fields>text_field tests to hoot

### PR DESCRIPTION
**Pull request by SimonGenin**
task 3705027

### CODE REVIEW

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        arch: '<form><field name="description"/></form>',
+    });
+    expect(".o_field_text textarea").toHaveCount(1);
+    expect(".o_field_text textarea").toHaveValue("Description as text");
+    await fieldTextArea("description").edit("New description");
+    expect(".o_field_text textarea").toHaveValue("New description");
+    await fieldTextArea("description").edit("Another description");
+    expect(".o_field_text textarea").toHaveValue("Another description");
+    await clickSave();
+    expect(".o_field_text textarea").toHaveValue("Another description");
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:41]: remove

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+});
+
+test("it has the correct height when edited", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:44]: textarea doesn't have a scrollbar when value is long

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    Product._records = [{ id: 1, description: "L\no\nn\ng\nD\ne\ns\nc\nr\ni\np\nt\ni\no\nn\n" }];
+    await mountView({
+        type: "form",
+        resModel: "product",
+        resId: 1,
+        arch: '<form><field name="description"/></form>',
+    });
+    const textarea = queryOne(".o_field_text textarea");
+    expect(textarea.clientHeight).toBe(textarea.scrollHeight - Math.abs(textarea.scrollTop));
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:53]: `expect(textarea.clientHeight).toBe(textarea.scrollHeight)`

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+});
+
+test("no vertical resize on edit", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:56]: remove

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+        type: "form",
+        resModel: "product",
+        resId: 1,
+        arch: `<form><field name="description"/><field name="name"/></form>`,
+    });
+    const textarea = queryOne(".o_field_text textarea");
+    const initialHeight = textarea.offsetHeight;
+    await fieldInput("name").edit("Let's trigger the onchange");
+    await animationFrame();
+    expect(textarea.offsetHeight).toBeGreaterThan(initialHeight);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:86]: check no scrollbar

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+    });
+    await contains(".o_data_row .o_data_cell").click();
+    const textarea = queryOne(".o_field_text textarea");
+    expect(textarea.clientHeight).toBe(textarea.scrollHeight);
+    await contains("tr:not(.o_data_row)").click();
+    const cell = queryOne(".o_data_row .o_data_cell");
+    expect(cell.clientHeight).toBe(cell.scrollHeight);
+});
+
+test("should resize on reset", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:108]: remove

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+test("set row on text fields", async () => {
+    Product._records = [{ id: 1, description: "Description as text" }];
+    await mountView({
+        type: "form",
+        resModel: "product",
+        resId: 1,
+        arch: `<form><field name="description" rows="4"/></form>`,
+    });
+    const textarea = queryOne(".o_field_text textarea");
+    expect(textarea.rows).toBe(4);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:137]: 2 fields, same value (one with rows="30"), height of second should be larger that first (+ check rows attr)

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+        resId: 1,
+        arch: `<form><sheet><group><field name="description"/></group></sheet></form>`,
+    });
+    expect(".o_field_text textarea").toHaveClass("o_field_translate");
+    expect(".o_field_text .btn.o_field_translate").toHaveCount(1);
+    await contains(".o_field_text .btn.o_field_translate").click();
+    expect(".modal").toHaveCount(1);
+});
+
+test("is translatable in create mode", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:177]: on new record

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+    });
+    await mountView({
+        type: "form",
+        resModel: "product",
+        arch: `<form><sheet><group><field name="description"/></group></sheet></form>`,
+    });
+    expect(".o_field_text .btn.o_field_translate").toHaveCount(1);
+});
+
+test("is translatable on notebook page", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:191]: remove

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 10]
```diff
@@ -0,0 +1,346 @@
...
+            <list editable="top">
+                <field name="description" />
+            </list>`,
+    });
+    await contains(".o_data_row .o_data_cell").click();
+    expect("textarea.o_input").toHaveCount(1);
+    expect("textarea.o_input").toHaveValue("Description as text");
+    expect(queryOne("textarea.o_input")).toBe(document.activeElement);
+    await fieldTextArea("description").press("Enter");
+    expect(queryOne("textarea.o_input")).toBe(document.activeElement);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:254]: check there's a new line in the textarea

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 11]
```diff
@@ -0,0 +1,346 @@
...
+    expect("tr.o_data_row").toHaveCount(1);
+});
+
+test("rendering in list view", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:258]: remove

Comment (aab-odoo) [reply addons/web/static/tests/views/fields/text_field.test.js:258]: move the old version (qunit) to list_view_tests

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 12]
```diff
@@ -0,0 +1,346 @@
...
+                </sheet>
+            </form>`,
+    });
+    expect(".o_popover .o_model_field_selector_popover").toHaveCount(0);
+    await press("alt+#");
+    await animationFrame();
+    expect(".o_popover .o_model_field_selector_popover").toHaveCount(1);
+});
+
+test("should correctly resize in o2m", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/text_field.test.js:312]: remove

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+            <list editable="top">
+                <field name="description" />
+            </list>`,
+    });
+    await contains(".o_data_row .o_data_cell").click();
+    expect("textarea.o_input").toHaveCount(1);
+    expect("textarea.o_input").toHaveValue("Description as text");
+    expect(queryOne("textarea.o_input")).toBe(document.activeElement);
+    expect("textarea.o_input").toHaveValue("Description as text");
+    await fieldTextArea("description").press("Enter");
```
Comment (SimonGenin) [inline addons/web/static/tests/views/fields/text_field.test.js:170]: Note: press("Enter") doesn't add \n in inputs. 
JUM has been made aware and will fix it in the near future.
So we don't test that we added the \n for now.

File: addons/web/static/tests/views/fields/text_field.test.js [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+    return contains(`.o_field_widget[name='${name}'] textarea`);
+}
+
+class Product extends models.Model {
+    description = fields.Text();
+}
+
+defineModels([Product]);
+
+onRpc("/web/dataset/call_kw/res.users/has_group", () => true);
```
Comment (SimonGenin) [inline addons/web/static/tests/views/fields/text_field.test.js:25]: just learned it could be a global hook so it's already a bit better.

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r- (will make the test work for mobile)

**Review by aab-odoo — APPROVED** [review]
Thanks @SimonGenin 

robodoo r+

**Pull request closed without merging**